api/github_api/util/SceGithubApiHandler.js

```diff
@@ -0,0 +1,142 @@
+const fetch = require('node-fetch');
+
+/**
+ * Handles website's backend request to interface with Github's various APIs
+ */
+class SceGithubApiHandler {
+  /**
+   * Create SceGithubApiHandler object
+   * @param {String} clientId Client ID generated from Github App
+   * @param {String} clientSecret Client secret generated from Github App
+   */
+  constructor(clientId, clientSecret) {
+    this.clientId = clientId;
+    this.clientSecret = clientSecret;
+  }
+
+  /**
+   * Converts Github date format to MM-DD-YYYY format
+   * @param {String} date Github's date string
+   * e.g. "2011-04-14T16:00:49Z"
+   * @returns {String} String of reformatted date
+   */
+  parseDate(date) {
+    const formattedDate = date.split('T')[0].split('-');
+    return `${formattedDate[1]}-${formattedDate[2]}-${formattedDate[0]}`;
+  }
+
+  /**
+   * Gets top 5 ranked users that have contributed to the given repo
+   * in the past 30 days
+   * @param {String} repoName Name of SCE Github repo
+   * @returns {Array<Object>|boolean} Array containing objects with
+   * 2 properties user and commit. Array is ordered by commit count
+   * e.g. [ {user: "user1", commits: 7}, {user: "user2", commits: 4}, ... ]
+   */
+  getContributorsInPastMonthFromRepo(repoName) {
+    return fetch(
+      `https://api.github.com/repos/SCE-Development/${repoName}/commits` +
+      `?client_id=${this.clientId}&client_secret=${this.clientSecret}`
+    )
+      .then(response => response.json())
+      .then(data => {
+        if(!data.length) {
+          return {};
+        }
+        const endDate = new Date(Date.now());
+        const startDate = new Date(endDate - (2592000000)); // Subtract 30 days
+        const commitCount = {};
+        const avatarUrls = {};
+        const contributors = [];
+
```

Comment:
Parse the data before you count it. 

You can actually [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) data before going into the foreach loop to avoid all these checks. 
Something like this...
```
data = data.filter( (commit) => (
    startDate <= Date(commit.commit.author.date)
    && Date(commit.commit.author.date) <= endDate
    && !commit.author.login.includes('[bot]')
```

Once the data has been filtered, use something like [groupby](https://www.geeksforgeeks.org/underscore-js-groupby-with-examples/) or [reduce](https://medium.com/@edisondevadoss/javascript-group-an-array-of-objects-by-key-afc85c35d07e) to organize the data by author. Once the objects have been grouped, you can just use the .length attribute of the newly generated author tag.

This makes the code more easy to understand in my opinion. It might be a bit difficult.

Reply:
great suggestion